app/views/layouts/devise.html.erb

```diff
@@ -21,12 +67,31 @@
               <%= sanitize notice, :tags => %w(br) %>
             </div>
           <% end %>
-          <%= yield %>
-        </main>
+
+          <%# Actual login form %>
+          <div class="pt-4">
+            <%= yield %>
+          </div>
+        </div>
       </div>
+
+      <%# Announcements %>
+      <% announcements = get_announcements() %>
+      <% if announcements.present? %>
```

Comment:
In case you implement the scope mentioned above, you would not need the announcements variable and could do
`if Announcement.active_on_main.any?`

Reply:
Just found [this](https://www.ombulabs.com/blog/benchmark/performance/rails/present-vs-any-vs-exists.html) concerning `.present?` vs. `.any?` vs. `.exists?` and therefore used `.exists?`.